Oauth

NiRO-bb opened the pull request:
Форму пока не успел сделать
Регистрация
* После успешной регистрации редирект на /auth/oauth, чтобы сгенерировать JWT, с которым уже можно доступ получить ко всему остальному
БД
* В таблицу с основными данными пользователя (service_user) добавил столбец hasGoogleReg для проверки наличия регистрации через гугл.
* Для хранения userinfo.profile используется новая таблица (user_info). Хранится email как первичный ключ и два поля с инфой: given_name и family_name.

NiRO-bb committed:
chore: add dependencies

# Conflicts:
#	pom.xml

NiRO-bb committed:
feat: add db migrations

update service_user table
add user_info table

NiRO-bb committed:
feat: add UserInfo DTO

add DTO, service and repository

NiRO-bb committed:
feat: update SecurityConfig

add Oauth2Login

NiRO-bb committed:
feat: add google exception

add GoogleRegistrationMissingException for authentication

NiRO-bb committed:
feat: update Auth

add signInOauth to AuthController
add signInOauth and signUpOauth to AuthService
add AuthUtil class
add TokenWithUser class

NiRO-bb committed:
chore: update JwtService

NiRO-bb committed:
fix: update User DTO

add isGoogleReg field to User
add findByEmail method to UserRepository

NiRO-bb committed:
test: update tests

NiRO-bb committed:
fix: update password encoder

NiRO-bb committed:
docs: update README

AlKl1M commented on src/main/java/com/example/AuthService/Service/AuthService.java:
```diff
@@ -44,8 +54,8 @@ public User signUp(User user) {
         boolean isExist = userRepo.existsByLoginOrEmail(user.getLogin(), user.getEmail());
         if (!isExist) {
             user.setPassword(passwordEncoder.encode(user.getPassword()));
+            System.out.println(user.getLogin());
```
Логи, не System.out. Хотя подозреваю что ты дебажил и просто забыл.

NiRO-bb replied:
Да, для проверки использовал. Забыл убрать

NiRO-bb committed:
chore: update dependencies

add thymeleaf

NiRO-bb committed:
feat: add custom login page

NiRO-bb committed:
fix: catch DataAccessException

replace RuntimeException with DataAccessException

NiRO-bb committed:
test: update tests

replace RuntimeException with DataAccessException

NiRO-bb committed:
fix: update AuthService

return default password (previously encoded password was returned)

NiRO-bb commented:
Добавил форму + исправил косяки

AlKl1M commented on src/main/java/com/example/AuthService/Service/JwtService.java:
```diff
@@ -17,14 +19,16 @@
-    Long expiration;
+    private Long expiration;
 
     public String getUsername(String token) {
```
У тебя в JwtService есть deprecated - нет смысла это юзать тем более в сервисе, где тебе не ограничивают версии. В коде еще есть такие места - поправь.

AlKl1M reviewed (changes requested):
По тестам норм, по функционалу тоже, но очень неудобно проверять - дели условно файл свойств на дев версию и на прод к примеру. В прод вкладывай то, что у тебя идет в контейнер, а дев - для локального развертывания + описывай все параметры - мне руками token.secret.key впихивать неудобно - вставляй туда значения .env, не надо свои токены кидать для этого, а выяснять чего не хватает - долго. Плюс не очень могу понять - ты запускаешь локально в intellij код или тестишь тестами/в контейнере? 

Плюс обязательно Docker-compose.yaml, сейчас при проверке мне надо поднимать только постгрю, а дальше кафка итд - делай заранее файл.

NiRO-bb commented:
> Плюс не очень могу понять - ты запускаешь локально в intellij код или тестишь тестами/в контейнере?

Локально запускаю. Docker только в тестах используется для test containers

NiRO-bb committed:
fix: update dependencies

NiRO-bb committed:
fix: update AuthService

return default password (previously encoded password was returned)

NiRO-bb committed:
chore: update system properties

add environment dependencies

NiRO-bb committed:
build: add docker

add dockerfile and docker_compose
add .env files to .gitignore

NiRO-bb committed:
docs: update README

NiRO-bb commented:
Добавил docker_compose.
Вместо разных .properties для dev и prod используются разные .env-файлы - .env_dev и .env_prod соответственно. Вроде пока и для dev, и для prod одни и те же переменные используются, но значения могут быть разными, поэтому сделал разные .env вместо разных .properties.
Список переменных, которые надо определить в .env можно в README посмотреть (сами .env_prod и .env_dev добавил в .gitignore)

AlKl1M reviewed (approved):
Мне вроде норм
Если у Артема будут замечания - фиксом.

NiRO-bb committed:
build: update .properties

add profile for dev - local deployment
add profile for prod - container deployment

NiRO-bb committed:
test: update dependencies

NiRO-bb committed:
build: update docker-compose

NiRO-bb committed:
docs: update README.md
